test_mcg_namespace_object_versions_crd[bucketclass_dict0] is constantly failing on 4.18

Relevant runs: 

https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/41633/1923156/1923273/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/41667/1923638/1923757/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42553/1957761/1958087/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42556/1959057/1959237/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42596/1960564/1960738/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42598/1960915/1961089/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42602/1961948/1962122/log

Backtrace : 
self = <test_mcg_namespace_s3_ops_crd.TestMcgNamespaceS3OperationsCrd object at 0x7f2377aaead0>
mcg_obj = <ocs_ci.ocs.resources.mcg.MCG object at 0x7f237af88d10>
cld_mgr = <ocs_ci.ocs.resources.cloud_manager.CloudManager object at 0x7f237c91d1d0>
bucket_factory = <function bucket_factory_fixture.._create_buckets at 0x7f2377c202c0>
bucketclass_dict = {'interface': 'OC', 'namespace_policy_dict': {'namespacestore_dict': {'aws': [(1, 'eu-central-1')]}, 'type': 'Single'}}


@pytest.mark.parametrize(
    argnames=["bucketclass_dict"],
    argvalues=[
        pytest.param(
            {
                "interface": "OC",
                "namespace_policy_dict": {
                    "type": "Single",
                    "namespacestore_dict": {"aws": [(1, "eu-central-1")]},
                },
            }
        ),
    ],
)
def test_mcg_namespace_object_versions_crd(
    self, mcg_obj, cld_mgr, bucket_factory, bucketclass_dict
):
    """
    Test object versioning S3 operations on namespace buckets/resources(CRDs).
    Validates put, get, delete object version operations

    """
    obj_versions = []
    version_key = "ObjKey-" + str(uuid.uuid4().hex)
    total_versions = 10
    aws_s3_resource = boto3.resource(
        "s3",
        endpoint_url=constants.MCG_NS_AWS_ENDPOINT,
        aws_access_key_id=cld_mgr.aws_client.access_key,
        aws_secret_access_key=cld_mgr.aws_client.secret_key,
    )

    ns_buc = bucket_factory(
        amount=1,
        interface=bucketclass_dict["interface"],
        bucketclass=bucketclass_dict,
    )[0]
    ns_bucket = ns_buc.name
    namespace_res = ns_buc.bucketclass.namespacestores[0].uls_name
    aws_s3_client = aws_s3_resource.meta.client

    # Put, Get bucket versioning and verify
    logger.info(f"Enabling bucket versioning on resource bucket: {namespace_res}")

  assert bucket_utils.s3_put_bucket_versioning(
        s3_obj=mcg_obj,
        bucketname=namespace_res,
        status="Enabled",
        s3_client=aws_s3_client,
    ), "Failed: PutBucketVersioning"

tests/functional/object/mcg/lifecycle/test_mcg_namespace_s3_ops_crd.py:482: 



ocs_ci/ocs/bucket_utils.py:1440: in s3_put_bucket_versioning
    return s3_client.put_bucket_versioning(
venv/lib64/python3.11/site-packages/botocore/client.py:595: in _api_call
    return self._make_api_call(operation_name, kwargs)
venv/lib64/python3.11/site-packages/botocore/context.py:123: in wrapper
    return func(args, *kwargs)



self = <botocore.client.S3 object at 0x7f237adb97d0>
operation_name = 'PutBucketVersioning'
api_params = {'Bucket': 'aws-uls-584e87645992424e88acc4402539a6c6', 'VersioningConfiguration': {'Status': 'Enabled'}}


@with_current_context()
def _make_api_call(self, operation_name, api_params):
    operation_model = self._service_model.operation_model(operation_name)
    service_name = self._service_model.service_name
    history_recorder.record(
        'API_CALL',
        {
            'service': service_name,
            'operation': operation_name,
            'params': api_params,
        },
    )
    if operation_model.deprecated:
        logger.debug(
            'Warning: %s.%s() is deprecated', service_name, operation_name
        )
    request_context = {
        'client_region': self.meta.region_name,
        'client_config': self.meta.config,
        'has_streaming_input': operation_model.has_streaming_input,
        'auth_type': operation_model.resolved_auth_type,
        'unsigned_payload': operation_model.unsigned_payload,
    }

    api_params = self._emit_api_params(
        api_params=api_params,
        operation_model=operation_model,
        context=request_context,
    )
    (
        endpoint_url,
        additional_headers,
        properties,
    ) = self._resolve_endpoint_ruleset(
        operation_model, api_params, request_context
    )
    if properties:
        # Pass arbitrary endpoint info with the Request
        # for use during construction.
        request_context['endpoint_properties'] = properties
    request_dict = self._convert_to_request_dict(
        api_params=api_params,
        operation_model=operation_model,
        endpoint_url=endpoint_url,
        context=request_context,
        headers=additional_headers,
    )
    resolve_checksum_context(request_dict, operation_model, api_params)

    service_id = self._service_model.service_id.hyphenize()
    handler, event_response = self.meta.events.emit_until_response(
        f'before-call.{service_id}.{operation_name}',
        model=operation_model,
        params=request_dict,
        request_signer=self._request_signer,
        context=request_context,
    )

    if event_response is not None:
        http, parsed_response = event_response
    else:
        maybe_compress_request(
            self.meta.config, request_dict, operation_model
        )
        apply_request_checksum(request_dict)
        http, parsed_response = self._make_request(
            operation_model, request_dict, request_context
        )

    self.meta.events.emit(
        f'after-call.{service_id}.{operation_name}',
        http_response=http,
        parsed=parsed_response,
        model=operation_model,
        context=request_context,
    )

    if http.status_code >= 300:
        error_info = parsed_response.get("Error", {})
        error_code = error_info.get("QueryErrorCode") or error_info.get(
            "Code"
        )
        error_class = self.exceptions.from_code(error_code)

      raise error_class(parsed_response, operation_name)
E           botocore.exceptions.ClientError: An error occurred (PermanentRedirect) when calling the PutBucketVersioning operation: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.



venv/lib64/python3.11/site-packages/botocore/client.py:1058: ClientError

--- a/tests/functional/object/mcg/lifecycle/test_mcg_namespace_s3_ops_crd.py
+++ b/tests/functional/object/mcg/lifecycle/test_mcg_namespace_s3_ops_crd.py
@@ -467,7 +467,7 @@ class TestMcgNamespaceS3OperationsCrd(E2ETest):
         )[0]
         ns_bucket = ns_buc.name
         namespace_res = ns_buc.bucketclass.namespacestores[0].uls_name
-        aws_s3_client = cld_mgr.aws_client.s3_client
+        aws_s3_client = cld_mgr.aws_client.client.meta.client
 
         # Put, Get bucket versioning and verify
         logger.info(f"Enabling bucket versioning on resource bucket: {namespace_res}")

--- a/tests/functional/object/mcg/lifecycle/test_mcg_namespace_s3_ops_crd.py
+++ b/tests/functional/object/mcg/lifecycle/test_mcg_namespace_s3_ops_crd.py
@@ -1,7 +1,6 @@
 import logging
 import uuid
 
-import boto3
 import pytest
 
 from ocs_ci.framework.pytest_customization.marks import (
@@ -461,17 +460,6 @@ class TestMcgNamespaceS3OperationsCrd(E2ETest):
         obj_versions = []
         version_key = "ObjKey-" + str(uuid.uuid4().hex)
         total_versions = 10
-        aws_s3_resource = boto3.resource(
-            "s3",
-            endpoint_url=constants.MCG_NS_AWS_ENDPOINT.format(
-                bucketclass_dict["namespace_policy_dict"]["namespacestore_dict"]["aws"][
-                    0
-                ][1]
-            ),
-            aws_access_key_id=cld_mgr.aws_client.access_key,
-            aws_secret_access_key=cld_mgr.aws_client.secret_key,
-        )
-
         ns_buc = bucket_factory(
             amount=1,
             interface=bucketclass_dict["interface"],
@@ -479,7 +467,7 @@ class TestMcgNamespaceS3OperationsCrd(E2ETest):
         )[0]
         ns_bucket = ns_buc.name
         namespace_res = ns_buc.bucketclass.namespacestores[0].uls_name
-        aws_s3_client = aws_s3_resource.meta.client
+        aws_s3_client = cld_mgr.aws_client.s3_client
 
         # Put, Get bucket versioning and verify
         logger.info(f"Enabling bucket versioning on resource bucket: {namespace_res}")

